walletconnectv2/src/main/kotlin/com/walletconnect/walletconnectv2/client/WalletConnectClient.kt

```diff
@@ -226,4 +255,20 @@ object WalletConnectClient {
         fun onSessionApproved(approvedSession: WalletConnect.Model.ApprovedSession)
         fun onSessionRejected(rejectedSession: WalletConnect.Model.RejectedSession)
     }
+
+    private fun safeCall(method: () -> Unit) {
+        try {
+            method()
+        } catch (e: WalletConnectExceptions) {
+            throw WalletConnect.WalletConnectError(e.message)
+        }
+    }
+
+    private fun safeCallWithReturn(method: () -> String?): String? {
+        try {
+            return method()
+        } catch (e: WalletConnectExceptions) {
+            throw WalletConnect.WalletConnectError(e.message)
+        }
+    }
```

Comment:
You can consolidate these into a single function with a generic return type

```suggestion
    private fun safeCall(method: () -> T): T {
        try {
            return method()
        } catch (e: WalletConnectExceptions) {
            throw WalletConnect.WalletConnectError(e.message)
        }
    }
```

Reply:
Since we agreed that exception goes to the core module and are used across all layers, it's not needed.